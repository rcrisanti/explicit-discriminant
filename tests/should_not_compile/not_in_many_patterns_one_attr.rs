use explicit_discriminant::ExplicitDiscriminant;

#[derive(ExplicitDiscriminant)]
#[pattern(..=-1, 4, 7 | 9..=12, 15..17, 19..)]
enum Test {
    VeryNegative = -9999999,
    LittleNegative = -1,
    Zero = 0,
    Four = 4,
    Five = 5,
    Seven = 7,
    Nine = 9,
    Ten = 10,
    Eleven = 11,
    Twelve = 12,
    Fifteen = 15,
    Sixteen = 16,
    Nineteen = 19,
    Eighteen = 18,
    VeryPositive = 123921380,
}

fn main() {
    let t = Test::Zero;
}
