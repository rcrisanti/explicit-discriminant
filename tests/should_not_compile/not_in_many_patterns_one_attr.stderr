error: discriminant does not match any pattern
 --> tests/should_not_compile/not_in_many_patterns_one_attr.rs:8:12
  |
8 |     Zero = 0,
  |            ^

error: discriminant does not match any pattern
  --> tests/should_not_compile/not_in_many_patterns_one_attr.rs:10:12
   |
10 |     Five = 5,
   |            ^

error: discriminant does not match any pattern
  --> tests/should_not_compile/not_in_many_patterns_one_attr.rs:19:16
   |
19 |     Eighteen = 18,
   |                ^^
